### DISCUSSION TITLE
MQTT message with last push scheduler

### DISCUSSION THREAD

**Discussion by simonepittis** [gmeks/myUplinkSmartConnect#11] [Ideas]
Hi, for you is it possible to have an MQTT message with the last push Scheduler? like used in Nordpool (see img1)?

This makes me help to integrate a view in my Home Assistant like Nordpool tomorrow energy price (see img2 and img3)

img1:
![image](https://user-images.githubusercontent.com/92569194/228587842-99c29517-256c-4948-920c-a85225b2aac8.png)

img2:
![image](https://user-images.githubusercontent.com/92569194/228588024-2dafc6b4-a782-4d6e-9138-9d760a9656d7.png)

img3:
![image](https://user-images.githubusercontent.com/92569194/228588220-3bf5c29d-113f-49a0-ae4d-7f3dc75b68e1.png)

**Comment by gmeks** [comment]
Do you mean the price information? 

If you checked the source code, you will see that i have seperated the project to 2 different "projects". One does price fetching and one talks to water heater.

My general idea was price information should be integrated into homeassistant, esp the annoying Norwegian "support" system, to allow us to get effective price we are paying. 
Is that what you aiming after?

**Comment by simonepittis** [comment]
no no the energy price, I need to add the view of the scheduler for today and tomorrow of my Water heater.

like this for example:
![image](https://user-images.githubusercontent.com/92569194/228735734-67bc4473-35ac-47d6-b0b6-39dd61756c23.png)

And your APP knows this data because push it to MyUplink site.

I need only to have the same push in MQTT, get it and create the view in my Home Assistant

**Comment by gmeks** [comment]
Ok,

My thinking is to make a MQTT message with the new schedule, is there a spesific format that will make it easier to format in homeassistant? Or should i just use json

**Comment by simonepittis** [comment]
json it's ok :-)

**Comment by gmeks** [comment]
Are you building your own versions, there now is a MQTT message for Schedule that has the schedule?

Or should i release a new version

**Comment by simonepittis** [comment]
if you develop the MQTT message, since you know your own code better :-)

**Comment by gmeks** [comment]
[linux-arm64.zip](https://github.com/gmeks/myUplinkSmartConnect/files/11216672/linux-arm64.zip)
[linux-x64.zip](https://github.com/gmeks/myUplinkSmartConnect/files/11216673/linux-x64.zip)
[windows.zip](https://github.com/gmeks/myUplinkSmartConnect/files/11216674/windows.zip)

This version sends MQTT message Schedule that contains the planned schedule, il do a full release later when its run a while

**Comment by simonepittis** [comment]
Tusen takk :-)

I will test it this evening

**Comment by simonepittis** [comment]
Hi, tested now... but I don't see any message in the MQTT

![Screenshot 2023-04-16 at 17 07 12](https://user-images.githubusercontent.com/92569194/232322148-263326b4-badc-41ae-a334-87024f0f4d79.png)

![Screenshot 2023-04-16 at 17 05 17](https://user-images.githubusercontent.com/92569194/232322168-5c0d3737-6ab3-4f66-90fd-0f37ffd65774.png)

**Reply by simonepittis** [reply]
my bad... works :-)

good Job takk

**Comment by simonepittis** [comment]
Topic: heater/18760xxxxxxxxxxxxxxxx0571/ScheduleQoS: 0

[{"Time":"2023-04-17T00:00:00+02:00","MaximumPower":200,"TargetTemprature":70},{"Time":"2023-04-16T17:17:26.562708+02:00","MaximumPower":200,"TargetTemprature":70},{"Time":"2023-04-16T23:17:26.5627307+02:00","MaximumPower":0,"TargetTemprature":50},{"Time":"2023-04-17T05:17:26.562736+02:00","MaximumPower":70,"TargetTemprature":50},{"Time":"2023-04-16T17:17:26.5627396+02:00","MaximumPower":200,"TargetTemprature":70},{"Time":"2023-04-16T23:17:26.5627421+02:00","MaximumPower":0,"TargetTemprature":50},{"Time":"2023-04-17T05:17:26.5627445+02:00","MaximumPower":70,"TargetTemprature":50},{"Time":"2023-04-16T17:17:26.5627473+02:00","MaximumPower":200,"TargetTemprature":70},{"Time":"2023-04-16T23:17:26.5627496+02:00","MaximumPower":0,"TargetTemprature":50},{"Time":"2023-04-17T05:17:26.5627517+02:00","MaximumPower":70,"TargetTemprature":50},{"Time":"2023-04-16T17:17:26.5627541+02:00","MaximumPower":200,"TargetTemprature":70},{"Time":"2023-04-16T23:17:26.5627564+02:00","MaximumPower":0,"TargetTemprature":50},{"Time":"2023-04-17T05:17:26.5627584+02:00","MaximumPower":70,"TargetTemprature":50},{"Time":"2023-04-16T17:17:26.5627606+02:00","MaximumPower":200,"TargetTemprature":70},{"Time":"2023-04-16T23:17:26.5627629+02:00","MaximumPower":0,"TargetTemprature":50},{"Time":"2023-04-17T05:17:26.5627649+02:00","MaximumPower":70,"TargetTemprature":50},{"Time":"2023-04-16T00:00:00+02:00","MaximumPower":200,"TargetTemprature":70}]

**Comment by gmeks** [comment]
Could you share what you did in homeassistant, perhaps others want to copy it

**Comment by simonepittis** [comment]
Hei ;-)

works but it's a draft now... I will build the code better and share all the config for all other users :-)
